boot/freeldr/freeldr/arch/i386/pcmem.c

```diff
@@ -286,6 +290,22 @@ PcMemGetBiosMemoryMap(PFREELDR_MEMORY_DESCRIPTOR MemoryMap, ULONG MaxMemoryMapSi
             break;
         }
 
+        /* Extra (input) safety. */
+        if (Regs.x.ecx < 20 || BiosMemoryMapSize < Regs.x.ecx)
```

Comment:
I'd prefer the constant (BiosMemoryMapSize) on the right side of the comparison. You're validating `Regs.x.ecx` so you want to know whether it's greater/less than something else, not if something else is less/greater than it.